feat: quickstart argilla-server

rhoitjadhav opened the pull request:
## Description

Quickstart argilla server using docker

Closes #2152

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

**How Has This Been Tested**
Tested locally

**Checklist**

- [ ] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [x] I added comments to my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

rhoitjadhav committed:
feat: quickstart argilla-server

dvsrepo commented on scripts/start_quickstart_argilla.sh:
```diff
@@ -0,0 +1,29 @@
+cat >users.yml <<EOF
+- username: "admin"
+  api_key: $USER_API_KEY
+  full_name: Hugging Face
```
For the quickstart, I would remove the Hugging Face references/naming. Also as the workspace will be reflected in the UI as `admin` let's call this user `team` for example so it's more friendly as a workspace name.

rhoitjadhav replied:
Its done

frascuchon commented on scripts/start_quickstart_argilla.sh:
```diff
@@ -0,0 +1,29 @@
+# Creating users.yml file
+cat >users.yml <<EOF
+- username: "admin"
+  api_key: $USER_API_KEY
```
We need 2 different credentials configuration:

1. The admin username, should be private to quickstart deployers
2. The argilla username that can be shared to login with.

We should have something like:

`ADMIN_PASSWORD`
`ADMIN_API_KEY`
`ARGILLA_PASSWORD`
`ARGILLA_API_KEY`

dvsrepo replied:
@rhoitjadhav keep in mind that these ENV vars should all have defaults, and these defaults should be aligned with the defaults we have when launching Argilla locally (user credentials and API key). This is highly important for the user experience of these quickstart users both for the local and HF deployment: we don't require them to set anything or generate passwords if they just want to quickly launch an Argilla instance for experimentation. We provide these ENVs to give more flexibility to more advanced quickstart users that want to better restrict access to datasets.

For the API_KEYS:
`ADMIN_API_KEY`: this is the one that should be used if you want to keep a secret API key for uploading data. This should be then mapped with the API_KEY of HF Spaces deployments.
`ARGILLA_API_KEY` should be hidden from public users of HF Spaces (i.e., not visible in the code for launching the Space) otherwise random users would be able to add datasets to the Space.

For PASSWORDS:
The role of the ARGILLA user is to be able to login into the UI and explore/label data. The ARGILLA user must have access to the workspace where datasets with `ADMIN_API_KEY` are uploaded.
The credentials should be the default ones we have currently for local installations. If quickstart users want to restrict access to random users they can set the `ARGILLA_PASSWORD` as a secret. 

For `ADMIN_PASSWORD` I don't have any comments, and for the ADMIN_USERNAME either. I think the most important thing is to keep it a no-brainer to launch a default quickstart instance.

frascuchon commented on quickstart.Dockerfile:
```diff
@@ -0,0 +1,30 @@
+RUN python3.9 get-pip.py
+
+# Install argilla
+RUN pip install argilla[server]
```
This will install the latest published release. We should have something similar to the `release.Dockerfile` image. Copy the generated wheels packages and install them locally.

dvsrepo replied:
Yes, this image should reuse/resemble as much as possible the other image.

rhoitjadhav replied:
Sure I'll use that

frascuchon commented on quickstart.Dockerfile:
```diff
@@ -0,0 +1,30 @@
+FROM ubuntu:20.04
```
Did you try with an official python image as a base image? Maybe make sense if it works, since could simplify image maintenance.

rhoitjadhav replied:
I think I should, actually for elasticsearch 8.5 its giving error (unrecognized service) when we try to start the elasticsearch so I'm thinking to change the image to Python 3.9 or may be I'll use ES 8.5 image and test the same.

frascuchon replied:
Okay. Let me know if you have some troubles

frascuchon commented on quickstart.Dockerfile:
```diff
@@ -0,0 +1,30 @@
+FROM ubuntu:20.04
+
+# Exposing ports
+EXPOSE 6900
```
It could be also interesting expose the default elasticsearch port (9200)

rhoitjadhav replied:
Sure I'll add that

frascuchon commented on quickstart.Dockerfile:
```diff
@@ -0,0 +1,30 @@
+RUN curl -fsSL https://artifacts.elastic.co/GPG-KEY-elasticsearch | apt-key add -
+RUN echo "deb https://artifacts.elastic.co/packages/7.x/apt stable main" | tee -a /etc/apt/sources.list.d/elastic-7.x.list
+RUN apt update
+RUN apt -y install elasticsearch
```
Will this command install also the elasticsearch deps like java virtual machine and all the stuff?

rhoitjadhav replied:
Yes it does. This step is suggested in their [docs](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/deb.html#deb-repo) also.

frascuchon replied:
great!

Commit:
chore: Update version

leiyre committed:
feat: Include record event_timestamp (#2156)

# Description

Include record event_timestamp

Closes #1911

**Type of change**

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [ ] Documentation update

**How Has This Been Tested**

Please describe the tests that you ran to verify your changes. And
ideally reference `tests`.

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I added comments to my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works

rhoitjadhav committed:
feat: es 8.5.3 changes

rhoitjadhav committed:
feat: run argilla with default user if params not provided

Commit:
chore: Merge branch 'main' into develop

frascuchon commented on quickstart.Dockerfile:
```diff
@@ -0,0 +1,37 @@
+
+# Copy execution script
+COPY scripts/start_quickstart_argilla.sh /
+RUN chmod +x /start_quickstart_argilla.sh
```
Duplicated statement

rhoitjadhav replied:
Done

frascuchon commented on scripts/start_quickstart_argilla.sh:
```diff
@@ -0,0 +1,48 @@
+# Disable security in elasticsearch configuration
+sudo sed -i "s/xpack.security.enabled: true/xpack.security.enabled: false/g" /etc/elasticsearch/elasticsearch.yml
+sudo sed -i "s/cluster.initial_master_nodes/#cluster.initial_master_nodes/g" /etc/elasticsearch/elasticsearch.yml
+echo "cluster.routing.allocation.disk.threshold_enabled: false" | sudo tee -a /etc/elasticsearch/elasticsearch.yml
```
This message is shown when launching the container. Could we hide it?

rhoitjadhav replied:
Yes its hidden

frascuchon reviewed (changes requested):
I've run it locally and it's working.

The only things to remark on (and we need to review):
- After running the container
```bash
docker run -p 6900:6900 -p 9200:9200 \
-e ADMIN_PASSWORD=admin -e ARGILLA_PASSWORD=1234 \
-e ADMIN_API_KEY=admin -e ARGILLA_API_KEY=argilla \
--name argilla-quickstart \
-d clay-quickstart
```
The container cannot be stopped with a Ctrl+C

- Stopping the container from the Docker dashboard and then rerunning it makes it crash (and no info is raised in logs)

A one-shot execution can be enough, but we should solve problems for a better user experience.

frascuchon commented on scripts/start_quickstart_argilla.sh:
```diff
@@ -0,0 +1,48 @@
+# Disable security in elasticsearch configuration
+sudo sed -i "s/xpack.security.enabled: true/xpack.security.enabled: false/g" /etc/elasticsearch/elasticsearch.yml
+sudo sed -i "s/cluster.initial_master_nodes/#cluster.initial_master_nodes/g" /etc/elasticsearch/elasticsearch.yml
+echo "cluster.routing.allocation.disk.threshold_enabled: false" | sudo tee -a /etc/elasticsearch/elasticsearch.yml
```
Could we disable this as part of the docker image build, since we need setup it up just once?

rhoitjadhav committed:
feat: add load_data, default settings with 2 users

rhoitjadhav committed:
add nopasswd sudoers

dvsrepo committed:
Update README.md (#2202)

Commit:
chore: Merge branch 'main' into develop

frascuchon committed:
refactor(Client): Separate api and client modules (#2199)

# Description

- The `client` module contains the new `Argilla` class (old `Api`) 
- The `api` module set up the main argilla functions like `load` and
`log`.

Even if both modules shares docstring and method signature, I prefer to
handle them separately. This will simplify when argilla client instances
are exposed for better integrations.

Closes #2194 

**Type of change**

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I added comments to my code
- [x] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works

Co-authored-by: Francisco Aranda <francisco@recogn.ai>

rhoitjadhav committed:
feat: creating users.yml file, add quickstart docker build in github workflow

Commit:
Merge branch 'develop' into feat/quickstart-docker-image

rhoitjadhav committed:
feat: update job needs section

rhoitjadhav committed:
feat: added logging

rhoitjadhav committed:
feat: added docker hub description job

rhoitjadhav committed:
feat: update repository name

rhoitjadhav committed:
feat: update repository name

rhoitjadhav committed:
feat: add quickstart readme

rhoitjadhav committed:
feat: add user with hf permissions

rhoitjadhav committed:
feat: creating users.yml using tee command

rhoitjadhav committed:
feat: copying users.yml file from codebase

rhoitjadhav committed:
feat: add logging

rhoitjadhav committed:
feat: remove users.yml

rhoitjadhav committed:
feat: uncomment changing user

rhoitjadhav committed:
feat: uncomment commands

rhoitjadhav committed:
feat: uncomment commands

rhoitjadhav committed:
feat: using ES8.5 image

rhoitjadhav committed:
feat: using ES8.5 image

Commit:
ci: Combine docker build with matrix params

frascuchon reviewed (changes requested):
It looks good to me.

Just a comment to apply. We will use the next patch release to create the first version of this new image. This will simplify integrations with HF spaces. 

So please, change the target branch from `develop` to `main` (be sure to keep only changes applied in this branch when you change the branch, you should apply a commit cherry-pick for that)

dvsrepo commented on scripts/start_quickstart_argilla.sh:
```diff
@@ -0,0 +1,44 @@
+sleep 15
+
+# Load data
+if [ "$LOAD_DATA_ENABLE" == "true" ]; then
```
@rhoitjadhav as discussed. Let's change this variable to `LOAD_DATASETS` and use the following options:

1. `full`: it loads all current datasets from load_data.py
2. `single`: it loads just the `sst-sentiment-explainability` (that is until line 53 of `load_data.py`
3. `none`: it doesn't load any dataset

rhoitjadhav replied:
Pushed the changes and deployed on HF

dvsrepo replied:
💯 !

rhoitjadhav committed:
feat: LOAD_DATASETS as full, single, none

rhoitjadhav committed:
feat: remove unnecessary packages and improve dockerfile

rhoitjadhav committed:
feat: change sleep time to 20 sec

rhoitjadhav committed:
feat: reverting to previous changes

rhoitjadhav committed:
feat: wait time for ES to 30 sec

Commit:
chore: Merge branch 'main' into feat/quickstart-docker-image

Commit:
chore: Restore main changes

rhoitjadhav committed:
feat: add Environment Variables section

rhoitjadhav committed:
feat: add LOAD_DATASETS environment Variable
